examples/tutorials/mocking.md

```diff
@@ -0,0 +1,618 @@
+---
+title: "Testing in isolation with mocks"
+description: "Master the art of mocking in your unit tests. Learn how spies, stubs, fake time, and other Deno tools let you improve your code and confidence"
+url: /examples/mocking_tutorial/
+---
+
+# Testing in isolation with mocks
+
+This guide builds on the
+[basics of testing in Deno](/examples/testing_tutorial/) to focus specifically
+on mocking techniques that help you isolate your code during testing.
+
+## Table of Contents
+
+- [Introduction](#introduction)
+- [Spying](#spying)
+- [Stubbing](#stubbing)
+- [Faking time](#faking-time)
+- [Advanced mocking patterns](#advanced-mocking-patterns)
+- [Real-world example](#real-world-example)
+- [Summary](#summary)
+
+## Introduction
+
+For effective unit testing, you'll often need to mock dependencies. Mocking is a
+technique used in testing where you replace real dependencies with simulated
+versions that you can control. This is particularly useful when testing
+components that interact with external services, such as APIs or databases.
+
+Deno provides powerful mocking utilities through the standard library without
+requiring additional frameworks, making your tests more reliable and faster.
+
+### Spying
+
+In Deno, you can [`spy`](https://jsr.io/@std/testing/doc/mock#spying) on a
+function to test that it behaves as expected.
+
+By using spies, you can verify that your code interacts correctly with its
+dependencies without setting up complex infrastructure.
```

Comment:
Nit: I feel like test spies is less about testing if a function behaves as expected and more about being able to create dummy functions on the fly to assert that it was called with the right arguments and stuff.

Reply:
have updated